demos/DeterministicSamplingPlanning.cpp

```diff
@@ -0,0 +1,179 @@
+*   * Neither the name of the Rice University nor the names of its
+*     contributors may be used to endorse or promote products derived
+*     from this software without specific prior written permission.
+*
+*  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+*  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+*  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+*  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+*  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+*  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+*  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+*  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+*  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+*  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+*  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+*  POSSIBILITY OF SUCH DAMAGE.
+*********************************************************************/
+
+/* Author: Leonard Bruns */
+
+#include <ompl/base/spaces/RealVectorStateSpace.h>
+#include <ompl/geometric/SimpleSetup.h>
+#include <ompl/geometric/planners/prm/PRMstar.h>
+#include <ompl/util/PPM.h>
+#include <ompl/base/samplers/DeterministicStateSampler.h>
+#include <ompl/base/samplers/deterministic/HaltonSequence.h>
+
+#include <ompl/config.h>
+
+#include <boost/filesystem.hpp>
+#include <iostream>
+#include <memory>
+
+namespace ob = ompl::base;
+namespace og = ompl::geometric;
+
+class Plane2DEnvironment
+{
+public:
+
+    Plane2DEnvironment(const char *ppm_file)
+    {
+        bool ok = false;
+        try
+        {
+            ppm_.loadFile(ppm_file);
+            ok = true;
+        }
+        catch(ompl::Exception &ex)
+        {
+            OMPL_ERROR("Unable to load %s.\n%s", ppm_file, ex.what());
+        }
+        if (ok)
+        {
+            auto space(std::make_shared<ob::RealVectorStateSpace>());
+            space->addDimension(0.0, ppm_.getWidth());
+            space->addDimension(0.0, ppm_.getHeight());
+            maxWidth_ = ppm_.getWidth() - 1;
+            maxHeight_ = ppm_.getHeight() - 1;
+            ss_ = std::make_shared<og::SimpleSetup>(space);
+
+            // set state validity checking for this space
+            ss_->setStateValidityChecker([this](const ob::State *state) { return isStateValid(state); });
+            space->setup();
+            ss_->getSpaceInformation()->setStateValidityCheckingResolution(1.0 / space->getMaximumExtent());
+            ss_->setPlanner(std::make_shared<og::PRMstar>(ss_->getSpaceInformation()));
+
+            // set the deterministic sampler
+            // 2D space, no need to specify bases specifically
+            space->setStateSamplerAllocator(std::bind(&Plane2DEnvironment::allocateHaltonStateSamplerRealVector, this, std::placeholders::_1, 2, std::vector<unsigned int>{2,3}));
```

Comment:
This is the main difference with `demos/Point2DPlanning.cpp`. Rather than duplicating the code and creating a near-identical demo, can you just extend Point2DPlanning.cpp? Use an extra boolean flag for the constructor called, say, `useDeterministicSampling` to conditionally enable this. Use some basic command line parsing to allow the user to enable this via the command line.

Reply:
This is now implemented as requested.